2024-09-24 17:57:05,718 - ERROR - Exception on /products [GET]
Traceback (most recent call last):
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\flask\app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\flask\app.py", line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\flask\app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\flask\app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\flask_login\utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "G:\TITO\VS_CODE\ShambaSmart\website\views.py", line 293, in products
    return render_template('products.html', farm=farm, crops=crops)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\flask\templating.py", line 150, in render_template
    return _render(app, template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\flask\templating.py", line 131, in _render
    rv = template.render(context)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\jinja2\environment.py", line 1304, in render
    self.environment.handle_exception()
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\jinja2\environment.py", line 939, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "G:\TITO\VS_CODE\ShambaSmart\website\templates\products.html", line 1, in top-level template code
    {% extends 'base.html' %}
  File "G:\TITO\VS_CODE\ShambaSmart\website\templates\base.html", line 66, in top-level template code
    {% block content %}
  File "G:\TITO\VS_CODE\ShambaSmart\website\templates\products.html", line 126, in block 'content'
    <form method="POST" action="{{ url_for('views.add_sale') }}">
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\flask\app.py", line 1084, in url_for
    return self.handle_url_build_error(error, endpoint, values)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\flask\app.py", line 1073, in url_for
    rv = url_adapter.build(  # type: ignore[union-attr]
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\werkzeug\routing\map.py", line 924, in build
    raise BuildError(endpoint, values, method, self)
werkzeug.routing.exceptions.BuildError: Could not build url for endpoint 'views.add_sale'. Did you forget to specify values ['crop_id']?
2024-09-24 17:57:08,736 - ERROR - Exception on /products [GET]
Traceback (most recent call last):
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\flask\app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\flask\app.py", line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\flask\app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\flask\app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\flask_login\utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "G:\TITO\VS_CODE\ShambaSmart\website\views.py", line 293, in products
    return render_template('products.html', farm=farm, crops=crops)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\flask\templating.py", line 150, in render_template
    return _render(app, template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\flask\templating.py", line 131, in _render
    rv = template.render(context)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\jinja2\environment.py", line 1304, in render
    self.environment.handle_exception()
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\jinja2\environment.py", line 939, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "G:\TITO\VS_CODE\ShambaSmart\website\templates\products.html", line 1, in top-level template code
    {% extends 'base.html' %}
  File "G:\TITO\VS_CODE\ShambaSmart\website\templates\base.html", line 66, in top-level template code
    {% block content %}
  File "G:\TITO\VS_CODE\ShambaSmart\website\templates\products.html", line 126, in block 'content'
    <form method="POST" action="{{ url_for('views.add_sale') }}">
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\flask\app.py", line 1084, in url_for
    return self.handle_url_build_error(error, endpoint, values)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\flask\app.py", line 1073, in url_for
    rv = url_adapter.build(  # type: ignore[union-attr]
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\werkzeug\routing\map.py", line 924, in build
    raise BuildError(endpoint, values, method, self)
werkzeug.routing.exceptions.BuildError: Could not build url for endpoint 'views.add_sale'. Did you forget to specify values ['crop_id']?
2024-09-24 17:57:42,399 - ERROR - Exception on /products [GET]
Traceback (most recent call last):
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\flask\app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\flask\app.py", line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\flask\app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\flask\app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\flask_login\utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "G:\TITO\VS_CODE\ShambaSmart\website\views.py", line 293, in products
    return render_template('products.html', farm=farm, crops=crops)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\flask\templating.py", line 150, in render_template
    return _render(app, template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\flask\templating.py", line 131, in _render
    rv = template.render(context)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\jinja2\environment.py", line 1304, in render
    self.environment.handle_exception()
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\jinja2\environment.py", line 939, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "G:\TITO\VS_CODE\ShambaSmart\website\templates\products.html", line 1, in top-level template code
    {% extends 'base.html' %}
  File "G:\TITO\VS_CODE\ShambaSmart\website\templates\base.html", line 66, in top-level template code
    {% block content %}
  File "G:\TITO\VS_CODE\ShambaSmart\website\templates\products.html", line 127, in block 'content'
    <form method="POST" action="{{ url_for('views.add_sale') }}">
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\flask\app.py", line 1084, in url_for
    return self.handle_url_build_error(error, endpoint, values)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\flask\app.py", line 1073, in url_for
    rv = url_adapter.build(  # type: ignore[union-attr]
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "g:\TITO\VS_CODE\ShambaSmart\.venv\Lib\site-packages\werkzeug\routing\map.py", line 924, in build
    raise BuildError(endpoint, values, method, self)
werkzeug.routing.exceptions.BuildError: Could not build url for endpoint 'views.add_sale'. Did you forget to specify values ['crop_id']?
